consumer/consumer.go

```diff
@@ -327,3 +365,87 @@ func storeMetadataReceivedEvent(pw *writer.ParquetWriter, event ethereum.Metadat
 		log.Trace().Msg("Wrote metadata_received event to Parquet file")
 	}
 }
+
+func (c *Consumer) HandleValidatorMetadataEvent() error {
+	for {
+		select {
+		case event := <-c.validatorMetadataChan:
+			c.log.Trace().Any("event", event).Msg("Received validator event")
+
+			maddr, err := ma.NewMultiaddr(event.Multiaddr)
+			if err != nil {
+				c.log.Error().Err(err).Msg("Invalid multiaddr")
+				continue
+			}
+
+			ip, err := maddr.ValueForProtocol(ma.P_IP4)
+			if err != nil {
+				ip, err = maddr.ValueForProtocol(ma.P_IP6)
+				if err != nil {
+					c.log.Error().Err(err).Msg("Invalid IP in multiaddr")
+					continue
+				}
+			}
+
+			portStr, err := maddr.ValueForProtocol(ma.P_TCP)
+			if err != nil {
+				c.log.Error().Err(err).Msg("Invalid port in multiaddr")
+				continue
+			}
+
+			port, err := strconv.Atoi(portStr)
+			if err != nil {
+				c.log.Error().Err(err).Msg("Invalid port number")
+				continue
+			}
```

Comment:
It seems not the best way to do it, but it works for now. IP and Port can be added as fields to the NATS event.